raiden-ts/src/raiden.ts

```diff
@@ -338,8 +339,16 @@ export class Raiden {
     const network = await provider.getNetwork();
 
     // if no ContractsInfo, try to populate from defaults
-    if (!contracts) {
+    let contracts;
+    if (!contractsOrUserDepositAddress) {
       contracts = getContracts(network);
+    } else if (typeof contractsOrUserDepositAddress === 'string') {
+      // if an Address is provided, use it as UserDeposit contract address entrypoint and fetch
+      // all contracts from there
+      assert(Address.is(contractsOrUserDepositAddress), ErrorCodes.DTA_INVALID_ADDRESS);
```

Comment:
I'm too lazy to check this right now: would this lead to a generic error message as from the `errors.json` only or does it get extended with the name of the variable? Else this functions takes quite a lot of parameter. Telling him which one was wrong would be quite helpful. I think we do this at many other places.

Reply:
It's just a generic error. Sure, I'll include the details for this.